### PR TITLE
[Cleanup] change all direct print to LOG_* macro

### PR DESCRIPTION
**Pull request by ksaito7**
This PR changes all direct printf/cout calls and original logging schema to LOG_* macro for code cleanup. In addition, direct assert calls change to PL_ASSERT macro.

### CODE REVIEW

File: src/include/index/bwtree.h [hunk 1]
```diff
@@ -108,25 +101,7 @@ namespace bwtree {
...
-
-#else
-
-#define bwt_printf(fmt, ...)   \
-  do {                         \
-    dummy(fmt, ##__VA_ARGS__); \
-  } while (0);
-
-#endif
+static void dummy(const char *, ...) {}
```
Comment (apavlo) [inline src/include/index/bwtree.h]: Should we just get rid of this now?

File: src/include/index/bwtree.h [hunk 2]
```diff
@@ -346,127 +303,123 @@ class BwTreeBase {
    * This function must be called when the garbage pool is empty
    */
   void DestroyThreadLocal() {
-    bwt_printf("Destroy %lu thread local slots\n", thread_num);
-    
+    LOG_TRACE("Destroy %lu thread local slots\n", thread_num);
```
Comment (apavlo) [inline src/include/index/bwtree.h]: Remove the newline character after all of these new LOG_TRACE

File: src/include/index/bwtree.h (hunk 1 above)
Comment (ksaito7) [reply src/include/index/bwtree.h]: I think these should be deleted because these are used nowhere.

File: src/include/index/bwtree.h (hunk 2 above)
Comment (ksaito7) [reply src/include/index/bwtree.h]: Sorry I will fix it soon.

File: src/include/index/bwtree.h (hunk 1 above)
Comment (ksaito7) [reply src/include/index/bwtree.h]: I've deleted its dummy function.

**Comment by apavlo** [comment]
I resolved the conflicts by hand on Github but it doesn't look like that triggered the build.

**Comment by apavlo** [comment]
Rebased + merged manually.

**Pull request closed without merging**